### ISSUE TITLE
Memory Module Error

### ISSUE THREAD

**Issue by DerekYu177** [DerekYu177/425G14#29]
Running Test 4, the registers are assigned correctly, but the data memory is still empty. 

For reference, test 4 is the following :

```asm
addi $1, $0, 1 #place 1 into regsiter 1
addi $2, $0, 2
addi $3, $0, 3 #stall to avoid data hazard
sw $1, 0($0)  #store contents of reg 1 into memory address 0
```

**Comment by DerekYu177** [comment]
![image](https://cloud.githubusercontent.com/assets/11148779/24787417/28c8bba0-1b36-11e7-974d-4efe676a5262.png)

**Comment by DerekYu177** [comment]
To explain:
The flags are asserted on the falling edge, and so is the address. But reads only occur in the first half, so the RM is unable to see the request. 

Easiest possible fix: 
Move the read instruction half a CC earlier.

**Comment by DerekYu177** [comment]
Okay apparently windows notepad doesn't see spaces like atom does which explains why the instructions into the ID stage were staggered.

**Comment by DerekYu177** [comment]
Also, somehow `regwrite` is left asserted in a WB stage but not being purposefully asserted.

**Comment by DerekYu177** [comment]
Okay part of the problem is that the stages were active even when `initialization` was asserted. I reconnected these now

**Comment by DerekYu177** [comment]
#31 fixed the instruction input issue I was also hunting down.

**Comment by DerekYu177** [comment]
Sorry, #32 is the real fix for the instruction IO

**Comment by DerekYu177** [comment]
OMG, we need to reference the register at position 1, not position 0! I'm such a dumbass

```asm
addi $1, $0, 1 #place 1 into regsiter 1
addi $2, $0, 2
addi $3, $0, 3 #stall to avoid data hazard
sw $1, 0($1)  #store contents of reg 1 into memory address 1
```

**Comment by DerekYu177** [comment]
It's related to #35 so I will tackle that one first.

**Comment by DerekYu177** [comment]
Ugh, because the memory is byte-addressable, we have to say
```asm
addi $1, $0, 1 #place 1 into regsiter 1
addi $2, $0, 2
addi $2, $0, 3
addi $3, $0, 4 #stall to avoid data hazard
sw $0, 0($1)  #store contents of reg 1 into memory address 0
```

**Comment by DerekYu177** [comment]
Fixed in #34, #35
